File: DesignPatternDemo/Pattern/Command/CommandMain.m
//
//  Command.m
//  DesignPatternDemo
//
//  Created by ruantong on 2019/5/11.
//  Copyright © 2019 wupeng. All rights reserved.
//

#import "CommandMain.h"
#import "RequirementGroup.h"
#import "PageGroup.h"
#import "CodeGroup.h"

#import "Invoker.h"
#import "Command.h"
#import "AddRequirementCommand.h"
#import "DeletePageCommand.h"

@implementation CommandMain

- (void)main{
//    Group * rg = [RequirementGroup new];
//    [rg find];
//    [rg add];
//    [rg plan];
//
//    NSLog(@"------客户端要求删除一个节目------");
//    Group * pg = [PageGroup new];
//    [pg find];
//    [pg delete1];
//    [pg plan];
    
    Invoker * xiaosan = [Invoker new];
    NSLog(@"------客户端要求增加一项需求------");
    Command * command = [AddRequirementCommand new];
    [xiaosan setCommand:command];
    [xiaosan action];
    
    NSLog(@"------客户端要求删除一个节目------");
    Command * delcommand = [DeletePageCommand new];
    [xiaosan setCommand:delcommand];
    [xiaosan action];
}

@end

File: DesignPatternDemo/Pattern/Command/CommandMain.swift
//
//  CommandMain.swift
//  DesignPatternDemo
//
//  Created by wupeng on 2020/3/5.
//  Copyright © 2020 wupeng. All rights reserved.
//

import Foundation

class CommandMain: NSObject,IMain {
    func main() {
        print(("------------客户要求增加一项需求---------------"))
        let addCommand = AddRequirementCommand()
        var xiaoSan = Invoker.init(command: addCommand)
        xiaoSan.action()
    
        print("------------客户要求删除一个页面---------------")
        let delCommand = DeletePageCommand()
        xiaoSan = Invoker.init(command: delCommand)
        xiaoSan.action()
    }
}
